src/components/Questionnaire/QuestionnaireList.tsx

```diff
@@ -41,9 +57,175 @@ export function QuestionnaireList() {
+      <Table className="min-w-full divide-y divide-gray-200">
+        <TableHeader className="bg-gray-100 text-gray-700">
+          <TableRow>
+            <TableHead className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider">
+              {t("title")}
+            </TableHead>
+            <TableHead className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider">
+              {t("description")}
+            </TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody className="divide-y divide-gray-200 bg-white">
+          {questionnaireList?.length > 0 ? (
+            questionnaireList.map((questionnaire: QuestionnaireDetail) => (
+              <TableRow
+                key={questionnaire.id}
+                className="cursor-pointer hover:bg-gray-50"
+                onClick={() =>
+                  navigate(`/admin/questionnaire/${questionnaire.slug}`)
+                }
+              >
+                <TableCell className="px-6 py-2">
+                  {questionnaire.title && questionnaire.title.length > 20 ? (
+                    <TooltipProvider>
+                      <Tooltip>
+                        <TooltipTrigger>
+                          <span className="text-sm text-left font-semibold text-gray-950 truncate">
+                            {questionnaire.title}
+                          </span>
+                        </TooltipTrigger>
+                        <TooltipContent className="bg-black text-white z-40">
+                          {questionnaire.title}
+                        </TooltipContent>
+                      </Tooltip>
+                    </TooltipProvider>
+                  ) : (
+                    <span className="text-sm text-left font-semibold text-gray-950 truncate">
+                      {questionnaire.title}
+                    </span>
+                  )}
+                </TableCell>
+                <TableCell className="px-6 py-2">
+                  <div className="flex items-center justify-between space-x-4">
+                    <div className="truncate text-sm text-gray-900 break-words whitespace-normal">
+                      {questionnaire.description}
+                    </div>
+                    <Button
+                      variant="outline"
+                      size="sm"
+                      className="font-semibold shadow-gray-300 text-gray-950 border-gray-400"
+                    >
+                      <EyeIcon className="w-4 h-4 mr-0" />
+                      {t("View")}
+                    </Button>
+                  </div>
+                </TableCell>
+              </TableRow>
+            ))
+          ) : (
+            <TableRow>
+              <TableCell colSpan={3} className="text-center py-4">
+                {t("no_questionnaires_found")}
+              </TableCell>
+            </TableRow>
+          )}
+        </TableBody>
+      </Table>
+    </div>
+  );
+
```

Comment:
let's not define function inside a component and render it; react will not be able to memoize it properly; keep it as a component outside;